src/actions/LoginActions.ts

```diff
@@ -16,19 +16,22 @@ export enum LoginActionKeys {
 // synchronous action creators
 export const LoginActions = {
   loginRequest: createAction(LoginActionKeys.LOGIN_REQUEST),
-  loginExtend: createAction(LoginActionKeys.LOGIN_EXTEND, (token: Token, username: string, actualTimeOut: number) => ({
+  loginExtend: createAction(LoginActionKeys.LOGIN_EXTEND, (token: Token, username: string, currentTimeOut: number) => ({
     type: LoginActionKeys.LOGIN_EXTEND,
     token: token,
     username: username,
-    sessionTimeOut: actualTimeOut + config().session.extendedSessionTimeOut
-  })),
-  loginSuccess: createAction(LoginActionKeys.LOGIN_SUCCESS, (token: Token, username: string) => ({
-    type: LoginActionKeys.LOGIN_SUCCESS,
-    token: token,
-    username: username,
-    logged: true,
-    sessionTimeOut: new Date().getTime() + config().session.sessionTimeOut
+    sessionTimeOut: currentTimeOut + config().session.extendedSessionTimeOut
   })),
+  loginSuccess: createAction(
+    LoginActionKeys.LOGIN_SUCCESS,
+    (token: Token, username: string, currentTimeOut?: number) => ({
+      type: LoginActionKeys.LOGIN_SUCCESS,
+      token: token,
+      username: username,
+      logged: true,
+      sessionTimeOut: currentTimeOut || new Date().getTime() + config().session.sessionTimeOut
+    })
+  ),
   loginFailure: createAction(LoginActionKeys.LOGIN_FAILURE, (error: any) => ({
```

Comment:
I don't understand why you added currentTimeOut to the loginSuccess. In what case we are going to use this parameter in the login ?

Reply:
Mmm I thought that sessiontimeout was in redux-persist

Reply:
it is, but it's precisely overridden because of that "loginSuccess" call which resets the timer value.